### DISCUSSION TITLE
How to make Next.js page render on Client-side and totally ignore it on Server-side?

### DISCUSSION THREAD

**Discussion by bymoe** [vercel/next.js#15181] [Help] [answered]
Suppose I have `src/pages/video` page & I have many components on this page which require `window` and `document` and a lot of browser stuff and I don't benefit from Server-side rendering on this page, I just don't need it.

How to tell Next.js to Ignore this page on server to not prefetch or render it there.

**Comment by jamesmosier** [comment, marked as answer]
You can use [`next/dynamic`](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr) with `ssr: false`:

```js
const DynamicComponentWithNoSSR = dynamic(
  () => import('../components/hello3'),
  { ssr: false }
)
```

**Reply by bymoe** [reply]
Thank you @jamesgeorge007, How can I do this with entire pages not components?

**Reply by jamesmosier** [reply]
There is no way to do that for a page in particular. The easiest way would be to just use the page as your entry point and put all your logic in a component elsewhere:

```jsx
const MyNotSsrComponent = dynamic(
  () => import('../components/hello3'),
  { ssr: false }
)

export default function MyPage() {
  return <MyNotSsrComponent />
}
```

**Reply by bymoe** [reply]
Thank you!

**Reply by AVivero** [reply]
The following worked for me. 
```
function PageComponent() {...}

export default dynamic(() => Promise.resolve(PageComponent), {ssr: false})
```

**Comment by vasuneet** [comment]
I am not seeing component code if I follow this. How to confirm this is rendered on client side as I am just seeing HTML code not the ComponentName in my view source.

**Reply by berksafran** [reply]
If you want to understand whether client-side works or not, add `console.log("isSSR?")` to your component, like:

```
const Footer = () => {
  console.log("Is SSR? Footer -->", typeof window === "undefined");

  return (
    <footer>Footer</footer>
  );
};

export default Footer;
```

If you see any console.log in your terminal, it's not working as CSR. If you don't see, CSR works well. Also, with this code `typeof window === "undefined"` returns `true` on your terminal, if SSR works. (**not at your browser console**)
